backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -19,22 +31,149 @@ public class ReleaseService implements ReleaseAnnouncementUseCase {
+    this.deleteQueuedNormsPort = deleteQueuedNormsPort;
+    this.deleteUnpublishedNormPort = deleteUnpublishedNormPort;
   }
 
+  /**
+   * Queue all norms that are affected by the given announcement for publishing.
+   * This includes generating a new manifestation of the amending norm, and new expressions for each time boundary of
+   * every target norm. It also creates a new manifestation for the current expression of the target norm so it contains
+   * the information about the future changes and the correct "nachfolgende-version-id".
+   * <p></p>
+   * Previous releases of the same announcement of the current date are replaced by this release. Also, new
+   * unpublished manifestations are created for the amending norm and the current expression of the target norms to
+   * enable further work on the announcement.
+   * <p></p>
+   * NOTE: This is currently always creating new expressions for every time boundary (so what is necessary for the
+   * prätext-abgabe). When only metadata has changed to the previous abgabe no new expressions should be generated. This
+   * is not handled by this method yet.
+   *
+   * @param query The query specifying the {@link Announcement} to be loaded.
+   * @return The information about the announcement published.
+   */
   @Override
+  @Transactional
   public Announcement releaseAnnouncement(ReleaseAnnouncementUseCase.Query query) {
     var announcement = loadAnnouncementByNormEliUseCase.loadAnnouncementByNormEli(
       new LoadAnnouncementByNormEliUseCase.Query(query.eli())
     );
-    announcement.setReleasedByDocumentalistAt(Instant.now());
 
+    // find all norms to publish
+    Set<Norm> normsToPublish = new HashSet<>();
+    normsToPublish.add(normService.loadNorm(new LoadNormUseCase.Query(announcement.getEli())));
+
+    for (Norm norm : normsToPublish) {
+      var analysis = norm.getMeta().getAnalysis();
+      if (analysis.isEmpty()) {
+        break;
+      }
+
+      analysis
+        .get()
+        .getActiveModifications()
+        .stream()
+        .map(TextualMod::getDestinationHref)
+        .flatMap(Optional::stream)
+        .map(Href::getExpressionEli)
+        .flatMap(Optional::stream)
+        .map(eli -> normService.loadNorm(new LoadNormUseCase.Query(eli)))
+        .forEach(normsToPublish::add);
+    }
+
+    // Delete the files from a previous release of the same announcement if they are still queued for publishing.
+    // This is to prevent residual files from remaining if some time boundaries changed and this release will create
+    // different expressions.
+    normsToPublish.forEach(norm ->
+      deleteQueuedNormsPort.deleteQueuedForPublishNorms(
+        new DeleteQueuedNormsPort.Command(norm.getWorkEli())
+      )
+    );
+
+    // generate all future versions for all norms that will be published
+    Set<Norm> allVersionsOfAllNormsToPublish = new HashSet<>();
+    for (Norm norm : normsToPublish) {
```

Comment:
No future version needed for the amending law right? And it is also within the set normsToPublish right?

Reply:
yes there wouldn't be any future versions created for the amending norm. So for amending norms just the `latestNormExpression` will be added to the set.

The normsToPublish include just one version of each norm that is published (so in the normal case the amending norm and the zf0 of the target norm). While `allVersionsOfAllNormsToPublish` contains all the different expressions and manifestations of those norms that will be published. So at the end `allVersionsOfAllNormsToPublish` contains the xmls that will be published.